File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/ResearchSubject.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class ResearchSubject : DomainResource, IResource
{
    public required Base.ResourceReference Study { get; set; }
    public Base.Period? Period { get; set; }
    public required string Status { get; set; }
    public Base.ResourceReference? Consent { get; set; }
    public string? ActualArm { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public required Base.ResourceReference Individual { get; set; }
    public string? AssignedArm { get; set; }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/ResearchSubject.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Period
from base import DomainResource
from base import Reference
from base import Identifier

@dataclass(kw_only=True)
class ResearchSubject(DomainResource):
    study: Reference
    period: Optional[Period] = None
    status: str
    consent: Optional[Reference] = None
    actual_arm: Optional[str] = None
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    individual: Reference
    assigned_arm: Optional[str] = None